lib/in_out/reader.py

```diff
@@ -2,39 +2,47 @@
 
     Returns:
-        content (log_data) : list of lists contaning log data for every day
-
+       logs : a dictionary with keys as a python date object and value as the data associated with that date
     """
-    log_data = []
-    for folder_iterator in range(starting_month, ending_month + 1):
-        extra0 = "0" if folder_iterator < 10 else ""
-        for file_iterator in range(starting_date if folder_iterator == starting_month else 1, ending_date + 1 if folder_iterator == ending_month else 32):
-            extra0_2 = "0" if file_iterator < 10 else ""
-            file_path = log_directory + extra0 + \
-                str(folder_iterator) + "/" + extra0_2 + \
-                str(file_iterator) + "/" + channel_name + ".txt"
-            if not os.path.exists(file_path):
-                if not((folder_iterator == 2 and (file_iterator == 29 or file_iterator == 30 or file_iterator == 31)) or ((folder_iterator == 4 or folder_iterator == 6 or folder_iterator == 9 or folder_iterator == 11) and file_iterator == 31)):
-                    print "[Error | io/linuxInput] Path " + file_path + " doesn't exist"
-                continue
-            with open(file_path) as f:
-                # contents stores all the lines of the file channel_name
-                if config.DEBUGGER:
-                    print "Working on: " + file_path
-                log_data.append(f.readlines())
-               
-    return log_data
+    logs={} 
+    for year_iterator in range(starting_year,ending_year+1):  
+        for month_iterator in range(starting_month if year_iterator == starting_year else 1, ending_month + 1 if year_iterator == ending_year else 13):
+            extra0 = "0" if month_iterator < 10 else ""        
+            for day_iterator in range(starting_date if (month_iterator == starting_month and year_iterator == starting_year) else 1, ending_date + 1 if (month_iterator == ending_month and year_iterator==ending_year) else 32):
+                extra0_2 = "0" if day_iterator < 10 else ""
+                file_path = log_directory+str(year_iterator)+"/"+extra0 + \
+                    str(month_iterator) + "/" + extra0_2 + \
+                    str(day_iterator) + "/" + channel_name + ".txt"
+                if not os.path.exists(file_path):
+                    if not((month_iterator == 2 and (day_iterator == 29 or day_iterator == 30 or day_iterator == 31)) or ((month_iterator == 4 or month_iterator == 6 or month_iterator == 9 or month_iterator == 11) and day_iterator == 31)):
+                        print "[Error | io/linuxInput] Path " + file_path + " doesn't exist"
+                    continue
+                with open(file_path) as f:
+                    """ day_data stores all the lines of the file channel_name """
+                    if config.DEBUGGER:
+                        print "Working on: " + file_path
+                    day_data=f.readlines()             
```

Comment:
it will also be great if we can store some auxiliary information too with the data like `channel name`.
you could do `logs[date_key] = {"data":day_data, "channel":channel_name}